src/game_events/manager.hpp

```diff
@@ -55,21 +57,23 @@ class manager
 	manager& operator=(const manager&) = delete;
 
 	explicit manager();
-	void read_scenario(const config& scenario_cfg);
+	void read_scenario(const config& scenario_cfg, game_lua_kernel& lk);
 	~manager();
 
-	/** Create an event handler. */
-	void add_event_handler(const config& handler, bool is_menu_item = false);
+	/** Create an event handler from an [event] tag. */
+	void add_event_handler_from_wml(const config& handler, game_lua_kernel& lk, bool is_menu_item = false);
+	/** Create an empty event handler. Expects the caller to finish setting up the event. */
+	pending_event_handler add_event_handler_from_lua(const std::string& name, const std::string& id, bool repeat = false, bool is_menu_item = false);
 
 	/** Removes an event handler. */
 	void remove_event_handler(const std::string& id);
 
 	/** Gets an event handler by ID */
 	const handler_ptr get_event_handler_by_id(const std::string& id);
 
-	void add_events(const config::const_child_itors& cfgs, const std::string& type = std::string());
+	void add_events(const config::const_child_itors& cfgs, game_lua_kernel& lk, const std::string& type = std::string());
```

Comment:
Although `type` isn't new, please add a comment saying what it is.

Reply:
I don't really think that belongs here, but… it's the unit type that contains the event, if any.

Reply:
There's no hint in the current docs that this expects a unit type, so it definitely needs documentation.

Reply:
I don't disagree. It just has nothing whatsoever to do with this PR.

Reply:
So, was the check for `utils::might_contain_variables(name)` just dead code? I mean, there was a code path specifically for events that occurred when `gd` was `nullptr`, including a specific path which would allow ones that didn't need interpolation to still happen.